tests/convert_numbers_test.c

```diff
@@ -236,6 +262,94 @@ static void test_utils_floatToText(void)
     }
 }
 
+static void test_utils_floatToTextExponential(void)
+{
+    size_t i;
+    char res[24];
+    int len;
+    int compareLen;
+
+    for (i = 0 ; i < sizeof(floats)/sizeof(floats[0]); i++)
+    {
+        compareLen = (int)strlen(floats_exponential[i]);
+        len = utils_floatToText(floats[i], (uint8_t*)res, sizeof(res), true);
+
+        CU_ASSERT(len >= compareLen);
+        if (len)
+        {
+            CU_ASSERT_NSTRING_EQUAL(res, floats_exponential[i], compareLen);
+            if (strncmp(res, floats_exponential[i], compareLen))
+                printf("%zu \"%g\" -> fail (%*s)\n", i, floats[i], len, res);
+        }
+        else
+        {
+            printf("%zu \"%g\" -> fail\n", i, floats[i]);
+        }
+    }
+
+    /* Test when no significant digits */
+    double val = DBL_MIN/FLT_RADIX;
+    CU_ASSERT_NOT_EQUAL(val, 0);
+    len = utils_floatToText(val, (uint8_t*)res, 6, true);
+    CU_ASSERT(len == 3);
+    if(len)
```

Comment:
The problem with CU_ASSERT_FATAL is that it prevents later tests from running if it fails. I prefer to know everything that is broken rather than just the first thing. It can help with deciding how to fix it.

Reply:
Having just one such assertion per test function would help with that. Of course, this would imply another kind of boiler plate... Since the existing code base does not do it, I am OK with leaving it as it is and doing some rework later on, on the whole code base.